### PR TITLE
feat(cli): support wildcards in `--project` option

### LINKED ISSUE #4747: Allow wildcards in `--project` option
### Clear and concise description of the problem

When running tests with the `--project` filter option, it'd be great to be able to pass wildcards (e.g. run all test environments for a monorepo package):

```sh
vitest --project effect:*
```

### Suggested solution

Allow simple minimatch patterns in the `--project` option

### Alternative

_No response_

### Additional context

_No response_

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide/).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### PR DESCRIPTION
**Pull request by fenghan34**
### Description

- Closes #4747 

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. If the feature is substantial or introduces breaking changes without a discussion, PR might be closed.
- [x] Ideally, include a test that fails without this PR but passes with it.
- [x] Please, don't make changes to `pnpm-lock.yaml` unless you introduce a new test example.

### Tests
- [x] Run the tests with `pnpm test:ci`.

### Documentation
- [x] If you introduce new functionality, document it. You can run documentation with `pnpm run docs` command.

### Changesets
- [x] Changes in changelog are generated from PR name. Please, make sure that it explains your changes in an understandable manner. Please, prefix changeset messages with `feat:`, `fix:`, `perf:`, `docs:`, or `chore:`.

### CODE REVIEW

File: packages/vitest/src/node/core.ts [hunk 1]
```diff
@@ -163,7 +163,17 @@ export class Vitest {
     const projects = await this.resolveWorkspace(cliOptions)
     this.projects = projects
     this.resolvedProjects = projects
-    const filteredProjects = toArray(resolved.project)
+    const filteredProjects = toArray(resolved.project).reduce((acc, s) => {
+      if (s.includes('*')) {
+        const pattern = new RegExp(`^${s.replace(/\*/, '.*')}`)
```
Comment (hi-ogawa) [inline packages/vitest/src/node/core.ts:168]: Coincidentally playwright just released `--project` flag with wildcard in the latest 1.42.0 https://github.com/microsoft/playwright/releases/tag/v1.42.0

Looking at what playwright did in
- https://github.com/microsoft/playwright/pull/29537
  (see `wildcardPatternToRegExp` in `packages/playwright/src/runner/projectUtils.ts`)

technically we should also escape special characters in `s` first. I think we can just copy over whole `wildcardPatternToRegExp`?

Comment (fenghan34) [reply packages/vitest/src/node/core.ts:168]: Thanks for the advise, I'll take a look

File: packages/vitest/src/node/core.ts [hunk 2]
```diff
@@ -163,7 +163,17 @@ export class Vitest {
     const projects = await this.resolveWorkspace(cliOptions)
     this.projects = projects
     this.resolvedProjects = projects
-    const filteredProjects = toArray(resolved.project)
+    const filteredProjects = toArray(resolved.project).reduce((acc, s) => {
```
Comment (hi-ogawa) [inline packages/vitest/src/node/core.ts:166]: nit: filtering projects twice looks a bit odd. How about doing this in one-go like this?

```tsx
      const patterns = filteredProjects.map(s => wildcardPatternToRegExp(s));
      this.projects = this.projects.filter(project =>
        patterns.some(pattern =>
          project.getName().match(pattern)
        )
      );
```

Comment (fenghan34) [reply packages/vitest/src/node/core.ts]: Thanks! Already updated.

**Review by hi-ogawa — APPROVED** [review]
Thanks!

**Pull request merged**